libcloud/storage/drivers/rgw.py

```diff
@@ -0,0 +1,137 @@
+                                                  proxy_url, token,
+                                                  retry_delay,
+                                                  backoff)
+
+
+class S3RGWStorageDriver(S3StorageDriver):
+
+    def __init__(self, key, secret=None, secure=True, host=None, port=None,
+                 api_version=None, region=S3_RGW_DEFAULT_REGION, **kwargs):
+        if host is None:
+            raise LibcloudError('host required', driver=self)
+        self.name = 'Ceph RGW S3 (%s)' % (region)
+        self.ex_location_name = region
+        self.region_name = region
+
+        self.signature_version = str(kwargs.pop('signature_version',
+                                                DEFAULT_SIGNATURE_VERSION))
+
+        if self.signature_version not in ['2', '4']:
+            raise ValueError('Invalid signature_version: %s' %
+                             (self.signature_version))
+
+        if self.signature_version == '2':
+            self.connectionCls = S3RGWOutscaleConnectionAWS2
+        elif self.signature_version == '4':
+            self.connectionCls = S3RGWOutscaleConnectionAWS4
+        self.connectionCls.host = host
+        super(S3RGWStorageDriver, self).__init__(key, secret,
+                                                 secure, host, port,
+                                                 api_version, region,
+                                                 **kwargs)
+
+    def _ex_connection_class_kwargs(self):
+        kwargs = {}
+        kwargs['signature_version'] = self.signature_version
+        return kwargs
+
+
+class S3RGWOutscaleConnectionAWS4(S3RGWConnectionAWS4):
+    pass
+
+
+class S3RGWOutscaleConnectionAWS2(S3RGWConnectionAWS2):
+    pass
+
+
+class S3RGWOutscaleStorageDriver(S3RGWStorageDriver):
+
+    def __init__(self, key, secret=None, secure=True, host=None, port=None,
+                 api_version=None, region=S3_RGW_OUTSCALE_DEFAULT_REGION,
+                 **kwargs):
+        if region not in S3_RGW_OUTSCALE_HOSTS_BY_REGION:
+            raise LibcloudError('Unknown region (%s)' % (region), driver=self)
+
+        self.name = 'OUTSCALE Ceph RGW S3 (%s)' % (region)
+        self.ex_location_name = region
+        self.region_name = region
+        self.signature_version = str(kwargs.pop('signature_version',
+                                                DEFAULT_SIGNATURE_VERSION))
+
+        if self.signature_version not in ['2', '4']:
+            raise ValueError('Invalid signature_version: %s' %
+                             (self.signature_version))
+
+        if self.signature_version == '2':
+            self.connectionCls = S3RGWOutscaleConnectionAWS2
+        elif self.signature_version == '4':
+            self.connectionCls = S3RGWOutscaleConnectionAWS4
+
+        host = S3_RGW_OUTSCALE_HOSTS_BY_REGION[region]
+        self.connectionCls.host = host
+        super(S3RGWStorageDriver, self).__init__(key, secret,
```

Comment:
You should pass `S3RGWOutscaleStorageDriver` as the first argument to `super()`. This will fix pylint issue.